command/volume_delete.go

```diff
@@ -136,7 +136,31 @@ func (c *VolumeDeleteCommand) deleteCSIVolume(client *api.Client, volID string,
 		}
 	}
 
-	err := client.CSIVolumes().DeleteOpts(&api.CSIVolumeDeleteRequest{
+	// get a CSI volume that matches the given prefix or a list of all matches
+	// if an exact match is not found.
+	stub, possible, err := getByPrefix[api.CSIVolumeListStub]("volumes", client.CSIVolumes().List,
+		func(vol *api.CSIVolumeListStub, prefix string) bool { return vol.ID == prefix },
+		&api.QueryOptions{
+			Prefix:    volID,
+			Namespace: c.namespace,
+		})
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Could not find existing volume to delete: %s", err))
+		return 1
+	}
+	if len(possible) > 0 {
+		out, err := csiFormatVolumes(possible, false, "")
```

Comment:
I'm not seeing test coverage for this, and if it were handled incorrectly (a bug introduced later?), it could cause external persistent state to be lost.

Reply:
We don't have any CLI unit coverage for the `volume delete` command and very little meaningful coverage of the CSI HTTP APIs at all for that matter, because of the setup requirements. We pretty much just test the routing and that's about it; the functionality is being covered in the RPC handler tests. The juice isn't worth the squeeze for coverage here.